Finjinon/PhotoCaptureViewController.swift

```diff
@@ -65,15 +70,78 @@ open class PhotoCaptureViewController: UIViewController, PhotoCollectionViewLayo
 
         view.backgroundColor = UIColor.black
 
-        previewView = UIView(frame: view.bounds)
+        NotificationCenter.default.addObserver(forName: NSNotification.Name.UIDeviceOrientationDidChange, object: nil, queue: nil) { (_) -> Void in
+            switch UIDevice.current.orientation {
+            case .faceDown, .faceUp, .unknown:
+                ()
+            case .landscapeLeft, .landscapeRight, .portrait, .portraitUpsideDown:
+                self.orientation = UIDevice.current.orientation
+                self.updateWidgetsToOrientation()
+            }
+        }
+    }
+
+    private func updateImagePickerButton() {
+        if imagePickerAdapter == nil {
+            if pickerButton != nil {
+                pickerButton?.removeFromSuperview()
+                pickerButton = nil
+            }
+        } else {
+            if pickerButton == nil {
+                let pickerButtonWidth: CGFloat = 114
+                pickerButton = UIButton(frame: CGRect(x: viewFrame.width - pickerButtonWidth - buttonMargin, y: viewFrame.origin.y + buttonMargin, width: pickerButtonWidth, height: 38))
+                pickerButton!.setTitle(NSLocalizedString("Photos", comment: "Select from Photos buttont itle"), for: UIControlState())
```

Comment:
As long as you're here you might do a bit of boycotting and help out that comment. :-D

Reply:
Asblobrutly 🤓